charts/federation-v2/templates/statefulset.yaml

```diff
@@ -0,0 +1,39 @@
+---
```

Comment:
(Query), Not sure why we chose StatefulSet for controller-manager, maybe because that's the default. IMHO, We don't have any requirement for it to be a StatefulSet, instead Deployment would have worked better for HA. Is the reason behind StatefulSet because controller-manager installs CRDs? @marun wdyt?

Reply:
@marun @pmorie comments for this? I think we can change the federation v2 controller to deployment instead of sts, as we do not have state need to persist and also do not need to do discovery via headless service.

Reply:
The rational for StatefulSet (iaccording to feedback I received on `#kubebuilder`) was that deployments don't guarantee only a single instance will be running at a time, and running more than one controller (in the absence of leader election) could cause problems.

Reply:
> deployments don't guarantee only a single instance will be running at a time

Not clear for what does this mean, if set `replica` as 1, can not we guarantee only a single instance?

Reply:
My understanding is that while setting `replicas: 1` will ensure that eventually the number of replicas will converge on 1, when configuration is changed such that a redeploy occurs, there may be more than 1 instance running.  StatefulSets is more careful about ensuring only a single instance is running at a time.  I encourage you to confirm this perspective with the kubebuilder devs.